framework/PostProcessorFunctions/TypicalHistoryFromHistorySet.py

```diff
@@ -82,6 +83,14 @@ def run(self,inputDic):
     if self.outputLen is None:
       self.outputLen = np.asarray(inputDict['output'][inputDict['output'].keys()[0]][self.pivotParameter])[-1]
 
+    ## Check if data is synchronized
+    referenceHistory = inputDict['output'].keys()[0]
+    referenceTimeAxis = inputDict['output'][referenceHistory][self.pivotParameter]
+    for hist in inputDict['output']:
+      if (str(inputDict['output'][hist][self.pivotParameter]) != str(referenceTimeAxis)):
+        errorMessage = '{} Interfaced Post-Processor "{}": one or more histories in the historySet have different time scales (e.g., reference points: {} and {})'.format(self.__class__.__name__, self.name,referenceHistory, hist)
```

Comment:
This doesn't need to be changed, but it is worth noting that the error handling system should automatically be printing the class (or string equivalent type) who raises an error, so it is unnecessary (but fine) to repeat it in the message.  If that's not the case, we should fix that in the message handler.